src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -257,12 +257,28 @@ public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
 
             public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
             {
-                VisitList(node.LeftVariables);
+                Visit(node.Left);
                 Visit(node.Right);
                 // don't map the deconstruction, conversion or assignment steps
                 return null;
             }
 
+            public override BoundNode VisitForEachStatement(BoundForEachStatement node)
```

Comment:
In a `BoundForEachStatement`, the `DeconstructionOpt` is marked to skip in visitor.
` <Field Name="DeconstructionOpt" Type="BoundForEachDeconstructStep" Null="allow" SkipInVisitor="true"/>`

So in `BoundNodes.xml.Generated.cs` we get:
```C#
   public override BoundNode VisitForEachStatement(BoundForEachStatement node)
        {
            this.Visit(node.IterationVariableType);
            this.Visit(node.Expression);
            this.Visit(node.Body);
            return null;
        }
```